src/coreclr/src/binder/bindertracing.cpp

```diff
@@ -231,3 +271,178 @@ namespace BinderTracing
+                    static_assert(HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND) == COR_E_FILENOTFOUND,
+                                    "COR_E_FILENOTFOUND has sane value");
+
+                    result = Result::AssemblyNotFound;
+                    errorMsg.Set(s_assemblyNotFoundMessage);
+                    break;
+
+                case FUSION_E_APP_DOMAIN_LOCKED:
+                    result = Result::IncompatibleVersion;
+
+                    {
+                        const auto &reqVersion = m_attemptInfo.AssemblyNameObject->GetVersion();
+                        errorMsg.Printf(W("Requested version %d.%d.%d.%d is incompatible with found version"),
+                            reqVersion->GetMajor(),
+                            reqVersion->GetMinor(),
+                            reqVersion->GetBuild(),
+                            reqVersion->GetRevision());
+                        if (resultAssembly != nullptr)
+                        {
+                            const auto &foundVersion = resultAssembly->GetAssemblyName()->GetVersion();
+                            errorMsg.AppendPrintf(W(" %d.%d.%d.%d"),
+                                foundVersion->GetMajor(),
+                                foundVersion->GetMinor(),
+                                foundVersion->GetBuild(),
+                                foundVersion->GetRevision());
+                        }
+                    }
+                    break;
+
+                case FUSION_E_REF_DEF_MISMATCH:
+                    result = Result::MismatchedAssemblyName;
+                    errorMsg.Printf(W("Requested assembly name '%s' does not match found assembly name"), m_attemptInfo.AssemblyName.GetUnicode());
+                    if (resultAssembly != nullptr)
+                        errorMsg.AppendPrintf(W(" '%s'"), resultAssemblyName.GetUnicode());
+
+                    break;
+
+                default:
+                    if (SUCCEEDED(hr))
+                    {
+                        result = Result::Success;
+                        _ASSERTE(resultAssembly != nullptr);
+                        // Leave errorMsg empty in this case.
+                    }
+                    else
+                    {
+                        result = Result::Failure;
+                        errorMsg.Printf(W("Resolution failed with HRESULT (%08x)"), m_hr);
+                    }
+            }
+        }
+
+        FireEtwResolutionAttempted(
+            GetClrInstanceId(),
+            m_attemptInfo.AssemblyName,
+            static_cast<uint16_t>(stage),
+            m_attemptInfo.AssemblyLoadContext,
+            static_cast<uint16_t>(result),
+            resultAssemblyName,
+            resultAssemblyPath,
+            errorMsg);
+    }
+
+    // static
+    void ResolutionAttemptedOperation::TraceAppDomainAssemblyResolve(AssemblySpec *spec, PEAssembly *resultAssembly, Exception *exception)
+    {
+        Result result;
```

Comment:
This should not do any work if the tracing is not enabled.

Reply:
🤦‍♀